### PR TITLE
Shared instance identity + Connect to Imbue

### PR DESCRIPTION
**Pull request by andrewlaack**
Introduce a single shared per-instance Imbue credential that authenticates the instance to Imbue services, plus a Connect to Imbue flow. Reworked to use the DB settings table + first_boot.toml introduced in #249 (no config.toml mutation, no restart).

State management: the credential lives in the DB settings table (core/identity_store.py), read live like domains. Managed spaces seed it from first_boot.toml on first boot; non-managed spaces obtain it via Connect. cert-api reads the credential from the store, falling back to the deprecated cert_api_keycloak_* config fields for already-deployed instances.

Connect to Imbue: a Settings section sends the owner to Imbue to authorize; Imbue returns a one-time code, which the instance exchanges server-to-server and writes to the settings table. No config file mutation, no restart (read live). Verified live end to end on a real instance: the credential lands in the settings table and the status endpoint flips to connected with no restart.

Around 113 new tests plus 50+ live edge cases on a real instance; pre-commit and test CI green.

Depends on #249 (merged).

### CODE REVIEW

File: compute_space/src/compute_space/core/connect.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    ``instance_base_url`` is this instance's own https origin; the one-time code is
+    returned to it (at CONNECT_CALLBACK_PATH), so the callback must be on the
+    instance's own zone.
+    """
+    callback = f"{instance_base_url.rstrip('/')}{CONNECT_CALLBACK_PATH}"
+    query = urlencode({"zone": zone, "callback": callback})
+    return f"{frontend_base_url.rstrip('/')}{_CONNECT_START_PATH}?{query}"
+
+
+def exchange_code_for_credential(
```
Comment (zplizzi) [inline compute_space/src/compute_space/core/connect.py:50]: what's the one time code from?

File: compute_space/src/compute_space/core/connect.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        return KeycloakClientCredentials(
+            issuer_url=str(body["issuer_url"]),
+            client_id=str(body["client_id"]),
+            client_secret=str(body["client_secret"]),
+        )
+    except (ValueError, KeyError, TypeError) as e:
+        raise ConnectError("connect exchange returned a malformed response") from e
+
+
+def persist_instance_identity(config_path: str, credential: KeycloakClientCredentials) -> None:
```
Comment (zplizzi) [inline compute_space/src/compute_space/core/connect.py]: i've been trying to avoid making the config file mutable, i feel like that gets messy. can we store these in the db instead? the initial secret from our provisioning can go into the `first_boot.toml` samuel added here: https://github.com/imbue-openhost/openhost/pull/249, and he also added a `settings` table for mutable key/value pairs.

File: compute_space/src/compute_space/core/connect.py (hunk 1 above)
Comment (andrewlaack) [reply compute_space/src/compute_space/core/connect.py:50]: It's the single-use code we issue to an instance's callback after the owner authorizes in the frontend
